File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float input;
    int coins;
    int counter = 0; //initalizating counter
    //check the correct input
    do
    {
        input = get_float("Change owed:");
    }
    while (input < 0);
    coins = round(input * 100); //Get coins from float
    //Count number of coins
    while (coins >= 25)
    {
        coins -= 25;
        counter++;
    }
    while (coins >= 10)
    {
        coins -= 10;
        counter++;
    }
    while (coins >= 5)
    {
        coins -= 5;
        counter++;
    }
    while (coins >= 1)
    {
        coins--;
        counter++;
    }
    printf("%i\n", counter);
}

File: pset6/cash/cash.py
from cs50 import get_float

# array with nominal and conter to count coins
nominal = [.25, .10, .05, .01]
counter = 0

# I won't use float to integer transfer cause of it just works
# I'm using round function

money = get_float("Change owed: ")
# Check the correct input
while money < 0:
    money = get_float("Change owed: ")

# Loop which counts coins
#   ??? I remember that there is better path to solve this problem???
#   ??? But I can't find out it
for i in nominal:
    while money >= i:
        money = round(money - i, 2)
        counter += 1
print(counter)

